### PR TITLE
add generic jsonresponse

### LINKED ISSUE #43: add generic JSON response class
It looks like most routes which don't implement `ImgResponse` are manually setting cache-ttl response headers:

image

All of these routes (I think) are JSON responses, and should implement a similar `JSONResponse` object which automatically sets cache-control rather than defining within the scope of each request.  The `JSONResponse` object would be very similar to `ImgResponse`, but of course implementing a different protocol.

### PR DESCRIPTION
**Pull request by geospatial-jeff**
Closes #43

### CODE REVIEW

File: titiler/endpoints/cog.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from fastapi import APIRouter, Depends, Path, Query
 
 from starlette.requests import Request
-from starlette.responses import HTMLResponse, Response
+from starlette.responses import HTMLResponse, JSONResponse, Response
```
Comment (vincentsarago) [inline titiler/endpoints/cog.py:30]: I think this `JSONResponse` should come from `titiler.ressources.responses`

File: titiler/endpoints/cog.py [hunk 2]
```diff
@@ -405,6 +402,9 @@ async def cog_tilejson(
         if minzoom:
             center[-1] = minzoom
         tjson = {
+            "tilejson": "2.2.0",
+            "version": "1.0.0",
+            "scheme": "xyz",
```
Comment (vincentsarago) [inline titiler/endpoints/cog.py:407]: why do you need to add this, before it was taken care of by the model!

**Comment by vincentsarago** [comment]
thanks @geospatial-jeff 

Quick question, what is restricting us to define the default response to our custom JSONResponse at the app level?

File: titiler/ressources/responses.py [hunk 3]
```diff
@@ -13,21 +13,16 @@ class XMLResponse(Response):
...
-        """Image response."""
-        headers.update({"Content-Type": media_type})
-        if ttl:
-            headers.update({"Cache-Control": "max-age=3600"})
-        self.body = self.render(content)
-        self.status_code = 200
-        self.media_type = media_type
-        self.background = background
-        self.init_headers(headers)
+    def __init__(self, *args, ttl: int = 2600, **kwargs):
```
Comment (vincentsarago) [inline titiler/ressources/responses.py:16]: I think `ttl` should be set to `3600` here!

**Comment by vincentsarago** [comment]
Alright I gave this a though (and some search on internet)

IMO we should remove any custom thing from the endpoints (meaning we don't set custom response/header) but we do this at the application level using middleware

```python
@app.middleware("http")
async def header_middleware(request: Request, call_next):
    """Add custom header."""
    response = await call_next(request)
    response.headers["Cache-Control"] = "public,max-age=3600"
    return response
```

what do you think @geospatial-jeff

**Comment by geospatial-jeff** [comment]
Closing in favor of #61

**Pull request closed without merging**